### PR TITLE
[#469] Refactored to use annotation API

### PR DESCRIPTION
**Pull request by kthoms**

### CODE REVIEW

File: org.eclipse.xtend.maven.plugin/pom.xml [hunk 1]
```diff
@@ -22,7 +22,6 @@
 		<dependency>
 			<groupId>com.google.guava</groupId>
 			<artifactId>guava</artifactId>
-			<version>21.0</version>
```
Comment (cdietrich) [inline org.eclipse.xtend.maven.plugin/pom.xml]: ??????

File: org.eclipse.xtend.maven.plugin/pom.xml [hunk 2]
```diff
@@ -207,7 +210,6 @@
 			</plugin>
 			<plugin>
 				<artifactId>maven-surefire-plugin</artifactId>
-				<version>2.22.1</version>
```
Comment (cdietrich) [inline org.eclipse.xtend.maven.plugin/pom.xml]: is here afaik intentional

File: org.eclipse.xtend.maven.plugin/pom.xml (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtend.maven.plugin/pom.xml]: is this a bom related change?

File: releng/org.eclipse.xtend.maven.parent/pom.xml [hunk 3]
```diff
@@ -294,18 +317,18 @@
 	<distributionManagement>
 		<repository>
 			<id>local-repo</id>
-			<url>${gradleMavenRepo}</url>
+			<url>file:${root-dir}/build/maven-repository</url>
```
Comment (cdietrich) [inline releng/org.eclipse.xtend.maven.parent/pom.xml]: looks like tis is not working here http://services.typefox.io/open-source/jenkins/job/xtext-xtend/job/kt_issue469_mojo_refactoring/2/console

Comment (cdietrich) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: i think this has to do how different jenkins hand work dirs and root dir differently

Comment (cdietrich) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: thus gradleMavenRepo is passed from outside

Comment (cdietrich) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: same here https://ci.eclipse.org/xtext/job/xtext-xtend/job/kt_issue469_mojo_refactoring/2/console

File: org.eclipse.xtend.maven.plugin/pom.xml (hunk 1 above)
Comment (kthoms) [reply org.eclipse.xtend.maven.plugin/pom.xml]: yes

File: releng/org.eclipse.xtend.maven.parent/pom.xml (hunk 3 above)
Comment (kthoms) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: ok, will check that, and likely revert

Comment (kthoms) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: xtext-maven-plugin uses the normal local Maven repo, maybe we should do less magic also here

Comment (cdietrich) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: problem: xtext-core and co. these need to be found as well

Comment (cdietrich) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: and the stuff here try to address all these

Comment (cdietrich) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: @kthoms the problem is: root-dir might not point where you expect

**Comment by cdietrich** [comment]
build still fails

**Comment by cdietrich** [comment]
needs rebase and fix of tests

File: releng/org.eclipse.xtend.maven.parent/pom.xml [hunk 4]
```diff
@@ -251,13 +278,9 @@
...
 			<id>useJenkinsSnapshots</id>
 			<repositories>
-				<repository>
-					<id>local-repo</id>
-					<url>${gradleMavenRepo}</url>
-				</repository>
 				<repository>
 					<id>lsp4j-from-jenkins</id>
-					<url>${JENKINS_URL}/job/lsp4j/job/master/lastStableBuild/artifact/build/maven-repository/</url>
+					<url>http://services.typefox.io/open-source/jenkins/job/lsp4j/job/master/lastStableBuild/artifact/build/maven-repository/</url>
```
Comment (ArneDeutsch) [inline releng/org.eclipse.xtend.maven.parent/pom.xml]: Looks like we do hard wire this more to one specific server. Looks like the wrong direction for me. At least we should leave a comment why we hardwire and probably what to do in future!?

Comment (cdietrich) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: https://github.com/eclipse/xtext-core/issues/975

**Comment by cdietrich** [comment]
ping @kthoms

File: releng/org.eclipse.xtend.maven.parent/pom.xml (hunk 3 above)
Comment (kthoms) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: Reverted that to old state with the gradleMavenRepo

File: releng/org.eclipse.xtend.maven.parent/pom.xml (hunk 4 above)
Comment (kthoms) [reply releng/org.eclipse.xtend.maven.parent/pom.xml]: For lsp4j this is OK. It is bound to the TF Jenkins and not available on the others.

**Comment by kthoms** [comment]
Build working on JIPP & TF
https://ci.eclipse.org/xtext/job/xtext-xtend/job/kt_issue469_mojo_refactoring/5/
http://services.typefox.io/open-source/jenkins/job/xtext-xtend/job/kt_issue469_mojo_refactoring/5/

@ArneDeutsch please review again

**Pull request merged**